### PR TITLE
Update CVE intel module to use NVD CVE API v2.0

### PR DESCRIPTION
**Pull request by heryxpc**
Issue https://github.com/lyft/cartography/issues/1275

Update the CVE intel module to the new [API 2.0 for vulnerabilities](https://nvd.nist.gov/developers/vulnerabilities).

Followed the [API 2.0 workflow](https://nvd.nist.gov/developers/api-workflows), but kept the year sync for compatibility with previous module version. Added as well delay among API calls to keep among [rate limits established](https://nvd.nist.gov/developers/start-here#divRateLimits).

Finally, switched to the Cartography ORM [model](https://github.com/lyft/cartography/tree/master/cartography/models/core) adding a new `CVEFeed` node representing the `:RESOURCE` parent of all `CVE` nodes.

### CODE REVIEW

File: cartography/cli.py [hunk 1]
```diff
@@ -685,6 +693,13 @@ def main(self, argv: str) -> int:
         else:
             config.semgrep_app_token = None
 
+        # CVE feed config
+        if config.cve_api_key_env_var:
+            logger.debug(f"Reading NVD CVE API key environment variable {config.cve_api_key_env_var}")
```
Comment (AgoraSecurity) [inline cartography/cli.py:698]: Leaving API Keys in logs is not a good security practice. I suggest either to remove the key completely or at least remove / hide most of the characters.

**Review by AgoraSecurity — COMMENTED** [review]
Feedback on logs

File: cartography/cli.py (hunk 1 above)
Comment (heryxpc) [reply cartography/cli.py:698]: Agree. On this case it's not an API key that is being logged, but the ENV_VAR holding it to follow the same pattern from previous API key configurations.

**Review by AgoraSecurity — APPROVED** [review]
LGTM

**Pull request merged**